File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        def dfs(curr, idx):
            if idx == len(nums):
                return
            
            if sum(curr) > target:
                return 
            elif sum(curr) == target:
                res.append(list(curr))
                return
            
            for i in range(idx, len(nums)):
                curr.append(nums[i])
                dfs(curr, i)
                curr.remove(nums[i])

        dfs([], 0)

        return res

File: Data Structures & Algorithms/binary-search/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1;
        while (l <= r) {
            int index = (l + r) / 2;
            int curr = nums[index];
            if (curr == target) {
                return index;
            } else if (curr < target) {
                l = index + 1;
            } else {
                r = index - 1;
            }
        }
        return -1;
    }
}
